File: tests/java/src/main/java/com/escape/tests/cases/Case302IndirectHolderEscape.java
package com.escape.tests.cases;

import java.util.HashMap;
import java.util.Map;

/** Adversarial spoiler case 302: holder indirection escape. */
public class Case302IndirectHolderEscape {
    private static Holder shared;

    private static class Holder {
        Map<String, String> value;
    }

    public static String execute(String input) {
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", "indirect_holder_escape");
        payload.put("entity", "spoiler");
        payload.put("stage", "adversarial");
        payload.put("input", raw);
        payload.put("checksum", "j302:" + raw.length());

        Holder h = new Holder();
        h.value = payload;
        // ESCAPE: payload stored through global holder indirection.
        shared = h;
        return "ok";
    }
}

File: tests/rust/cases/case_302_indirect_holder_escape.rs
#![allow(unused)]

use std::collections::HashMap;
use std::sync::{Mutex, OnceLock};

#[derive(Clone)]
struct Holder {
    value: HashMap<String, String>,
}

static SHARED_HOLDER: OnceLock<Mutex<Option<Holder>>> = OnceLock::new();

pub fn case_302_indirect_holder_escape(input: String) -> String {
    let raw = if input.is_empty() { "sample".to_string() } else { input };
    let mut payload: HashMap<String, String> = HashMap::new();
    payload.insert("task".to_string(), "indirect_holder_escape".to_string());
    payload.insert("entity".to_string(), "spoiler".to_string());
    payload.insert("stage".to_string(), "adversarial".to_string());
    payload.insert("input".to_string(), raw.clone());
    payload.insert("checksum".to_string(), format!("r302:{}", raw.len()));

    let h = Holder { value: payload };
    // ESCAPE: payload retained through holder object in global slot.
    SHARED_HOLDER
        .get_or_init(|| Mutex::new(None))
        .lock()
        .expect("holder lock")
        .replace(h);
    "ok".to_string()
}
